app/models/ems_event.rb

```diff
@@ -146,6 +146,15 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
+      new_vm = ManageIQ::Providers::Redhat::InfraManager::Vm.create_from_event(event)
```

Comment:
@blomquisg @gberginc I am open to suggestions how to place and run this code. The whole point of this change is to create vm object prior to refresh.